interfaces/systemd/snippet.go

```diff
@@ -0,0 +1,63 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2016 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package systemd
+
+import (
+	"bytes"
+	"fmt"
+)
+
+// Snippet describes systemd services that interface wishes to create.
+// Identical services from all snippets are combined and ignored.
+type Snippet struct {
+	Services map[string]Service `json:"services,omitempty"`
+}
+
+// Service describes a single systemd service file
+type Service struct {
```

Comment:
Not yet  as it's not there at all, there' a wrappers package that generates app-specific service units. We can evolve this but I think there's no need to move it yet